src/liquid/components/ld-notification/ld-notification.tsx

```diff
@@ -0,0 +1,190 @@
+import '../../components' // type definitions for type checks and intelliSense
+import { Component, h, Host, Listen, Prop, State, Watch } from '@stencil/core'
+
+type Notification = {
+  type: 'info' | 'warn' | 'alert'
+  content: string
+  timeout?: number
+}
+
+const DEFAULT_NOTIFICATION_TIMEOUT = 6000
+const FADE_TRANSITION_DURATION = 200
+
+@Component({
+  tag: 'ld-notification',
+  styleUrl: 'ld-notification.css',
+  shadow: false,
+})
+export class LdNotification {
+  /**
+   * Notification placement within the screen.
+   */
+  @Prop() placement: 'top' | 'bottom' = 'top'
+
+  @State() queue: Notification[] = []
+  @State() queueDismissed: Notification[] = []
+
+  @State() dismissTimeout: NodeJS.Timeout | null
+
+  @State() fadeoutTimeouts: (NodeJS.Timeout | null)[] = []
+
+  @State() currentNotification?: Notification
+
+  @Watch('currentNotification')
+  updateDismissTimeout() {
+    clearTimeout(this.dismissTimeout)
+
+    if (!this.currentNotification) return
+    if (this.currentNotification.type === 'alert') return
+    if (this.currentNotification.timeout === 0) return
+
+    this.dismissTimeout = setTimeout(() => {
+      this.handleNotificationDismiss()
+    }, this.currentNotification.timeout || DEFAULT_NOTIFICATION_TIMEOUT)
+  }
+
+  @Listen('ldNotification', {
+    target: 'window',
+    passive: true,
+  })
+  handleNotification(ev: CustomEvent<Notification>) {
+    ev.stopImmediatePropagation()
+    const newNotification = ev.detail
+
+    // If the same notification is already in queue (same content, same type), ignore this notification.
+    const inQueue = this.queue.find(
+      (notification) =>
+        notification.content === newNotification.content &&
+        notification.type === newNotification.type
+    )
+    if (inQueue) return
+
+    // Insert by relevance, whith error notifications being more relevant than non-error notifications.
```

Comment:
I've been thinking about this too. On the one hand Warning should be considered more relevant to the user. On the other hand changing the order of notifications too often is not such a good idea actually because it may confuse the user. Imagine the user reading an info notification and then all of a sudden a warning pops up and covers the info notification. The the user needs to re-read the info notification after dismissing the warning. Not so nice. That is why I decided to leave this behaviour for error message only as they should show up the least often. Maybe we should have another prop that can change order?

Reply:
Thinking about it, another prop feels like over-engineered to me. There should be a default behaviour. Guess we would need some user testings to find out what's the best approach here.